sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -82,36 +84,155 @@ private DoFnSignatures() {}
     return signature;
   }
 
+  /**
+   * The global context for the class, for use in analysis.
+   *
+   * <p>It contains much of the information that eventually becomes part of the {@link
+   * DoFnSignature}, but in an intermediate state.
+   */
+  @VisibleForTesting
+  static class GlobalAnalysisContext {
+
+    private final Map<String, StateDeclaration> stateDeclarations = new HashMap<>();
```

Comment:
I have a reasonable suspicion that this could grow over time. I'll keep it "as though" it is a fully engineered class.